### PR TITLE
Issue 114: renamed whitelist option to allowList. this change still supports whitelist option

### PR DESCRIPTION
**Pull request by hidekysilva**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by jsumners — APPROVED** [review]
LGTM

**Review by mcollina — CHANGES_REQUESTED** [review]
Could you add back a test that verifies the `whitelist` alias?

File: index.d.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ export interface RateLimitPluginOptions {
   cache?: number;
   store?: FastifyRateLimitStoreCtor;
   whitelist?: string[] | ((req: FastifyRequest, key: string) => boolean);
```
Comment (zekth) [inline index.d.ts:44]: Please add:
```typescript
  /**
   * @deprecated Use `allowList` property
   */
```

**Review by zekth — CHANGES_REQUESTED** [review]

**Comment by hidekysilva** [comment]
I added the test which passed and the deprecated annotation. Now, one of the redis tests is failing even though it was passing before my last commit. Any suggestions?

File: index.d.ts (hunk 1 above)
Comment (hidekysilva) [reply index.d.ts:47]: good to go

**Review by Eomm — APPROVED** [review]
+ other's suggestions

**Review by zekth — APPROVED** [review]
lgtm

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**